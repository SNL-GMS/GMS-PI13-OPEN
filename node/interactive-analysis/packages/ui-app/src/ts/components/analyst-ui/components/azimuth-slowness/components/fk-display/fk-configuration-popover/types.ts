import { FkTypes } from '@gms/common-graphql';
import { CheckboxSearchListTypes } from '@gms/ui-core-components';
import { FkConfigurationWithUnits, FkUnits } from '~analyst-ui/components/azimuth-slowness/types';

export enum FkConfigurationPopoverPanel {
  DEFAULT,
  ADVANCED
}

export interface FkConfigurationPopoverState {
  openPanel: FkConfigurationPopoverPanel;
  fkUnits: FkUnits;
  normalizeWaveforms: boolean;
  mediumVelocity: number;
  maximumSlowness: number;
  numberOfPoints: number;
  channelCheckboxes: CheckboxSearchListTypes.CheckboxItem[];
  useVerticalChannelOffsets: boolean;
}
export interface FkConfigurationPopoverProps extends FkTypes.FkConfiguration {
  fkUnitDisplayed: FkUnits;
  applyFkConfiguration(configuration: FkConfigurationWithUnits);
  close();
}
