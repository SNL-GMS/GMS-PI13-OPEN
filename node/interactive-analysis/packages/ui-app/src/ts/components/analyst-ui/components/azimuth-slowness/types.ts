import {
  CommonTypes,
  ConfigurationTypes,
  EventTypes,
  FkTypes,
  ProcessingStationTypes,
  SignalDetectionTypes,
  WaveformTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { Client } from '@gms/ui-apollo';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { ChildProps, MutationFunction } from 'react-apollo';
import { FilterType, FkThumbnailSize } from './components/fk-thumbnail-list/fk-thumbnails-controls';
import { LeadLagPairs } from './constants';

export interface FkParams {
  windowParams: FkTypes.WindowParameters;
  frequencyPair: FkTypes.FrequencyBand;
}

export interface LeadLagPairAndString {
  leadLagPairs: LeadLagPairs;
  windowParams: FkTypes.WindowParameters;
}

export enum FkUnits {
  FSTAT = 'FSTAT',
  POWER = 'POWER'
}

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits extends FkTypes.FkConfiguration {
  fkUnitToDisplay: FkUnits;
}

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  client: Client;
  currentStageInterval: AnalystWorkspaceTypes.StageInterval;
  currentTimeInterval: CommonTypes.TimeRange;
  selectedSdIds: string[];
  openEventId: string;
  sdIdsToShowFk: string[];
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSelectedSdIds(ids: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  userInputFkFrequency: FkTypes.FrequencyBand;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  fkInnerContainerWidthPx: number;
  fkFrequencyThumbnails: Immutable.Map<string, FkTypes.FkFrequencyThumbnail[]>;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: MutationFunction<{}>;
  computeFkFrequencyThumbnails: MutationFunction<{}>;
  setWindowLead: MutationFunction<{}>;
  markFksReviewed: MutationFunction<{}>;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  ChildProps<AzimuthSlownessMutations> &
  ProcessingStationTypes.DefaultStationsQueryProps &
  ConfigurationTypes.UIConfigurationQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: ProcessingStationTypes.ProcessingStation[];
  eventsInTimeRange: EventTypes.Event[];
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: EventTypes.Event | undefined;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<string, EventTypes.FeaturePrediction[]>;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  featurePredictionsForDisplayedSignalDetection: EventTypes.FeaturePrediction[];
  distances: EventTypes.LocationToStationDistance[];
  // Redux state as props
  selectedSdIds: string[];
  defaultWaveformFilters: WaveformTypes.WaveformFilter[];
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  userInputFkFrequency: FkTypes.FrequencyBand;
  fkUnitForDisplayedSignalDetection: FkUnits;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  fkInnerContainerWidthPx: number;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  markFksForSdIdsAsReviewed(sdIds: string[]): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(fkInput: FkTypes.FkInput): void;
  changeUserInputFks(
    windowParams: FkTypes.WindowParameters,
    frequencyBand: FkTypes.FrequencyBand
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkUnits): void;

  // Redux functions
  setSelectedSdIds(sdIds: string[]): void;
  setSdIdsToShowFk(sdIds: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}
