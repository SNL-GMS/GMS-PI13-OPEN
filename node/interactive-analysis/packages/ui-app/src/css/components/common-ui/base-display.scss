.base-display {
  display: flex;
  position: relative;
  flex-direction: column;
  &.full-width-height {
    height: 100%;
    width: 100%;
    flex-grow: 1;
  }
}
