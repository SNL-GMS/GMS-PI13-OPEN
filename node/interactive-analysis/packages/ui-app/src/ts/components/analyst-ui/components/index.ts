export { ReduxApolloWaveformDisplay as WaveformDisplay } from './waveform-display';
export { Events } from './events';
export { History } from './history';
export { SignalDetections } from './signal-detections';
export { Workflow } from './workflow';
export { ReduxApolloMap as Map } from './map';
export { Location } from './location';
export { Magnitude } from './magnitude';

export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness/azimuth-slowness-container';
