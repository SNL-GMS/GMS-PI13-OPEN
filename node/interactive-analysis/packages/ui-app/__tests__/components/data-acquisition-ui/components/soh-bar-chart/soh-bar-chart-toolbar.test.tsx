import { CommonTypes, SohTypes } from '@gms/common-graphql';
import { AnalystConfiguration } from '@gms/common-graphql/lib/graphql/ui-configuration/types';
import { uuid } from '@gms/common-util';
import { DropdownItem } from '@gms/ui-core-components/lib/components/ui-widgets/toolbar/types';
import { uniqueId } from 'lodash';
import React from 'react';
import { BaseDisplayContext } from '../../../../../src/ts/components/common-ui/components/base-display';
import {
  Toolbar,
  ToolbarProps
} from '../../../../../src/ts/components/data-acquisition-ui/components/soh-bar-chart/soh-bar-chart-toolbar';
import { initialFiltersToDisplay } from '../../../../../src/ts/components/data-acquisition-ui/shared/toolbars/soh-toolbar';
import {
  makeLagSortingDropdown,
  SOHLagOptions
} from '../../../../../src/ts/components/data-acquisition-ui/shared/toolbars/soh-toolbar-items';
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');
uuid.asString = jest.fn().mockImplementation(uniqueId);

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe('Toolbar class', () => {
  const sortDropdown: DropdownItem = makeLagSortingDropdown(SOHLagOptions.LAG_HIGHEST, jest.fn());
  // see https://stackoverflow.com/questions/57805917/mocking-refs-in-react-function-component
  const mockUseRef = (obj: any) => () =>
    Object.defineProperty({}, 'current', {
      get: () => obj,
      // tslint:disable-next-line: no-empty
      set: () => {}
    });
  const ref: any = mockUseRef({ refFunction: jest.fn() });
  const station: SohTypes.UiStationSoh = {
    id: '1',
    uuid: '1',
    needsAcknowledgement: true,
    needsAttention: true,
    sohStatusSummary: undefined,
    stationGroups: [],
    statusContributors: [],
    time: undefined,
    stationName: '1',
    allStationAggregates: [],
    channelSohs: [
      {
        channelName: 'adsf',
        channelSohStatus: undefined,
        allSohMonitorValueAndStatuses: [
          {
            monitorType: SohTypes.SohMonitorType.LAG,
            value: 10,
            valuePresent: true,
            status: SohTypes.SohStatusSummary.GOOD,
            hasUnacknowledgedChanges: true,
            contributing: false,
            quietUntilMs: 1,
            thresholdBad: 3,
            thresholdMarginal: 3
          },
          {
            monitorType: SohTypes.SohMonitorType.LAG,
            value: 11,
            valuePresent: true,
            status: SohTypes.SohStatusSummary.GOOD,
            hasUnacknowledgedChanges: true,
            contributing: false,
            quietUntilMs: 1,
            thresholdBad: 3,
            thresholdMarginal: 3
          }
        ]
      },
      {
        channelName: 'adsf2',
        channelSohStatus: undefined,
        allSohMonitorValueAndStatuses: [
          {
            monitorType: SohTypes.SohMonitorType.LAG,
            value: 10,
            valuePresent: true,
            status: SohTypes.SohStatusSummary.GOOD,
            hasUnacknowledgedChanges: true,
            contributing: false,
            quietUntilMs: 1,
            thresholdBad: 3,
            thresholdMarginal: 3
          },
          {
            monitorType: SohTypes.SohMonitorType.LAG,
            value: 11,
            valuePresent: true,
            status: SohTypes.SohStatusSummary.GOOD,
            hasUnacknowledgedChanges: true,
            contributing: false,
            quietUntilMs: 1,
            thresholdBad: 3,
            thresholdMarginal: 3
          }
        ]
      }
    ]
  };
  const uiAnalystConfiguration: AnalystConfiguration = {
    logLevel: CommonTypes.LogLevel.info,
    acknowledgementQuietDuration: 0,
    defaultFilters: [],
    defaultNetwork: undefined,
    sohStationStaleTimeMS: 30000,
    sohStationGroupNames: [],
    redisplayPeriod: 5,
    reprocessingPeriod: 20,
    availableQuietDurations: [],
    // tslint:disable-next-line: no-magic-numbers
    sohHistoricalDurations: [300000, 900000],
    systemMessageLimit: 1000
  };
  const toolbarProps: ToolbarProps = {
    statusesToDisplay: initialFiltersToDisplay,
    setStatusesToDisplay: jest.fn(),
    sortDropdown,
    forwardRef: ref,
    isStale: false,
    station,
    monitorType: SohTypes.SohMonitorType.LAG,
    uiAnalystConfiguration
  };
  const toolbar = Enzyme.mount(
    <BaseDisplayContext.Provider
      value={{
        glContainer: { width: 150, height: 150 } as any,
        widthPx: 150,
        heightPx: 150
      }}
    >
      <Toolbar {...toolbarProps} />
    </BaseDisplayContext.Provider>
  );
  it('should be defined', () => {
    expect(Toolbar).toBeDefined();
  });
  it('should match snapshot', () => {
    expect(toolbar).toMatchSnapshot();
  });
});
