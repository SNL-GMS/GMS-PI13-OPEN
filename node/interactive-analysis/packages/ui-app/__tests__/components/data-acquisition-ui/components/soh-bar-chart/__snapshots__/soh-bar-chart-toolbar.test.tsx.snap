// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar class should match snapshot 1`] = `
<Component
  forwardRef={[Function]}
  isStale={false}
  monitorType="LAG"
  setStatusesToDisplay={[MockFunction]}
  sortDropdown={
    Object {
      "dropdownOptions": Object {
        "CHANNEL_FIRST": "Channel: A - Z",
        "CHANNEL_LAST": "Channel: Z - A",
        "LAG_HIGHEST": "Lag: highest to lowest",
        "LAG_LOWEST": "Lag: lowest to highest",
      },
      "label": "Sort Channels",
      "onChange": [MockFunction],
      "rank": 1,
      "tooltip": "Sort",
      "type": "Dropdown",
      "value": "Lag: highest to lowest",
      "widthPx": 220,
    }
  }
  station={
    Object {
      "allStationAggregates": Array [],
      "channelSohs": Array [
        Object {
          "allSohMonitorValueAndStatuses": Array [
            Object {
              "contributing": false,
              "hasUnacknowledgedChanges": true,
              "monitorType": "LAG",
              "quietUntilMs": 1,
              "status": "GOOD",
              "thresholdBad": 3,
              "thresholdMarginal": 3,
              "value": 10,
              "valuePresent": true,
            },
            Object {
              "contributing": false,
              "hasUnacknowledgedChanges": true,
              "monitorType": "LAG",
              "quietUntilMs": 1,
              "status": "GOOD",
              "thresholdBad": 3,
              "thresholdMarginal": 3,
              "value": 11,
              "valuePresent": true,
            },
          ],
          "channelName": "adsf",
          "channelSohStatus": undefined,
        },
        Object {
          "allSohMonitorValueAndStatuses": Array [
            Object {
              "contributing": false,
              "hasUnacknowledgedChanges": true,
              "monitorType": "LAG",
              "quietUntilMs": 1,
              "status": "GOOD",
              "thresholdBad": 3,
              "thresholdMarginal": 3,
              "value": 10,
              "valuePresent": true,
            },
            Object {
              "contributing": false,
              "hasUnacknowledgedChanges": true,
              "monitorType": "LAG",
              "quietUntilMs": 1,
              "status": "GOOD",
              "thresholdBad": 3,
              "thresholdMarginal": 3,
              "value": 11,
              "valuePresent": true,
            },
          ],
          "channelName": "adsf2",
          "channelSohStatus": undefined,
        },
      ],
      "id": "1",
      "needsAcknowledgement": true,
      "needsAttention": true,
      "sohStatusSummary": undefined,
      "stationGroups": Array [],
      "stationName": "1",
      "statusContributors": Array [],
      "time": undefined,
      "uuid": "1",
    }
  }
  statusesToDisplay={
    Map {
      "GOOD" => true,
      "BAD" => true,
      "MARGINAL" => true,
      "NONE" => true,
    }
  }
  uiAnalystConfiguration={
    Object {
      "acknowledgementQuietDuration": 0,
      "availableQuietDurations": Array [],
      "defaultFilters": Array [],
      "defaultNetwork": undefined,
      "logLevel": "info",
      "redisplayPeriod": 5,
      "reprocessingPeriod": 20,
      "sohHistoricalDurations": Array [
        300000,
        900000,
      ],
      "sohStationGroupNames": Array [],
      "sohStationStaleTimeMS": 30000,
      "systemMessageLimit": 1000,
    }
  }
>
  <div
    className="soh-drill-down__header"
  >
    <Component
      displayTimeWarning={false}
      leftItems={Array []}
      rightItems={
        Array [
          Object {
            "dropdownOptions": Object {
              "CHANNEL_FIRST": "Channel: A - Z",
              "CHANNEL_LAST": "Channel: Z - A",
              "LAG_HIGHEST": "Lag: highest to lowest",
              "LAG_LOWEST": "Lag: lowest to highest",
            },
            "label": "Sort Channels",
            "onChange": [MockFunction],
            "rank": 1,
            "tooltip": "Sort",
            "type": "Dropdown",
            "value": "Lag: highest to lowest",
            "widthPx": 220,
          },
        ]
      }
      setStatusesToDisplay={[Function]}
      sohStationStaleTimeMS={30000}
      statusFilterText="Filter by Status"
      statusesToDisplay={
        Map {
          "GOOD" => true,
          "BAD" => true,
          "MARGINAL" => true,
          "NONE" => true,
        }
      }
      toggleHighlight={[Function]}
      updateIntervalSecs={20}
      widthPx={150}
    >
      <Component
        items={
          Array [
            Object {
              "dropdownOptions": Object {
                "CHANNEL_FIRST": "Channel: A - Z",
                "CHANNEL_LAST": "Channel: Z - A",
                "LAG_HIGHEST": "Lag: highest to lowest",
                "LAG_LOWEST": "Lag: lowest to highest",
              },
              "label": "Sort Channels",
              "onChange": [MockFunction],
              "rank": 1,
              "tooltip": "Sort",
              "type": "Dropdown",
              "value": "Lag: highest to lowest",
              "widthPx": 220,
            },
            Object {
              "hasIssue": false,
              "label": "Last Updated",
              "rank": 2,
              "style": Object {
                "marginLeft": "1em",
              },
              "tooltip": "Most recent SOH data received",
              "tooltipForIssue": "SOH data has not updated in over 30 seconds",
              "type": "LabelValue",
              "value": "-",
              "widthPx": 400,
            },
            Object {
              "label": "Update Interval",
              "rank": 3,
              "tooltip": "Interval at which SOH data is processed",
              "type": "LabelValue",
              "value": "20 seconds",
              "widthPx": 400,
            },
          ]
        }
        itemsLeft={
          Array [
            Object {
              "colors": Map {
                "GOOD" => "#fff",
                "MARGINAL" => "#fff",
                "BAD" => "#fff",
                "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
              },
              "cyData": "filter-soh",
              "enumKeysToDisplayStrings": Map {
                "GOOD" => "Good",
                "MARGINAL" => "Marginal",
                "BAD" => "Bad",
                "NONE" => "None",
              },
              "enumOfKeys": Object {
                "BAD": "BAD",
                "GOOD": "GOOD",
                "MARGINAL": "MARGINAL",
                "NONE": "NONE",
              },
              "label": "Filter by Status",
              "menuLabel": "Filter by Status",
              "onChange": [Function],
              "onPopUp": [Function],
              "onPopoverDismissed": [Function],
              "rank": 1,
              "tooltip": "Filter by Status",
              "type": "CheckboxList",
              "values": Map {
                "GOOD" => true,
                "BAD" => true,
                "MARGINAL" => true,
                "NONE" => true,
              },
              "widthPx": 200,
            },
          ]
        }
        widthPx={150}
      >
        <ToolbarComponent
          items={
            Array [
              Object {
                "dropdownOptions": Object {
                  "CHANNEL_FIRST": "Channel: A - Z",
                  "CHANNEL_LAST": "Channel: Z - A",
                  "LAG_HIGHEST": "Lag: highest to lowest",
                  "LAG_LOWEST": "Lag: lowest to highest",
                },
                "label": "Sort Channels",
                "onChange": [MockFunction],
                "rank": 1,
                "tooltip": "Sort",
                "type": "Dropdown",
                "value": "Lag: highest to lowest",
                "widthPx": 220,
              },
              Object {
                "hasIssue": false,
                "label": "Last Updated",
                "rank": 2,
                "style": Object {
                  "marginLeft": "1em",
                },
                "tooltip": "Most recent SOH data received",
                "tooltipForIssue": "SOH data has not updated in over 30 seconds",
                "type": "LabelValue",
                "value": "-",
                "widthPx": 400,
              },
              Object {
                "label": "Update Interval",
                "rank": 3,
                "tooltip": "Interval at which SOH data is processed",
                "type": "LabelValue",
                "value": "20 seconds",
                "widthPx": 400,
              },
            ]
          }
          itemsLeft={
            Array [
              Object {
                "colors": Map {
                  "GOOD" => "#fff",
                  "MARGINAL" => "#fff",
                  "BAD" => "#fff",
                  "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
                },
                "cyData": "filter-soh",
                "enumKeysToDisplayStrings": Map {
                  "GOOD" => "Good",
                  "MARGINAL" => "Marginal",
                  "BAD" => "Bad",
                  "NONE" => "None",
                },
                "enumOfKeys": Object {
                  "BAD": "BAD",
                  "GOOD": "GOOD",
                  "MARGINAL": "MARGINAL",
                  "NONE": "NONE",
                },
                "label": "Filter by Status",
                "menuLabel": "Filter by Status",
                "onChange": [Function],
                "onPopUp": [Function],
                "onPopoverDismissed": [Function],
                "rank": 1,
                "tooltip": "Filter by Status",
                "type": "CheckboxList",
                "values": Map {
                  "GOOD" => true,
                  "BAD" => true,
                  "MARGINAL" => true,
                  "NONE" => true,
                },
                "widthPx": 200,
              },
            ]
          }
          minWhiteSpacePx={1}
          toolbarWidthPx={110}
          widthPx={150}
        >
          <div
            className="toolbar"
            style={
              Object {
                "width": "110px",
              }
            }
          >
            <div
              className="toolbar__left-group"
            >
              <div
                className="toolbar-item toolbar-item__left"
                key="1"
              >
                <Component
                  addToPopoverMap={[Function]}
                  hasIssue={false}
                  item={
                    Object {
                      "colors": Map {
                        "GOOD" => "#fff",
                        "MARGINAL" => "#fff",
                        "BAD" => "#fff",
                        "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
                      },
                      "cyData": "filter-soh",
                      "enumKeysToDisplayStrings": Map {
                        "GOOD" => "Good",
                        "MARGINAL" => "Marginal",
                        "BAD" => "Bad",
                        "NONE" => "None",
                      },
                      "enumOfKeys": Object {
                        "BAD": "BAD",
                        "GOOD": "GOOD",
                        "MARGINAL": "MARGINAL",
                        "NONE": "NONE",
                      },
                      "label": "Filter by Status",
                      "menuLabel": "Filter by Status",
                      "onChange": [Function],
                      "onPopUp": [Function],
                      "onPopoverDismissed": [Function],
                      "rank": 1,
                      "tooltip": "Filter by Status",
                      "type": "CheckboxList",
                      "values": Map {
                        "GOOD" => true,
                        "BAD" => true,
                        "MARGINAL" => true,
                        "NONE" => true,
                      },
                      "widthPx": 200,
                    }
                  }
                  key="1"
                >
                  <div
                    style={Object {}}
                  >
                    <PopoverButtonComponent
                      cyData="filter-soh"
                      key="1"
                      label="Filter by Status"
                      onClick={[Function]}
                      onPopoverDismissed={[Function]}
                      popupContent={
                        <CheckboxList
                          checkboxEnum={
                            Object {
                              "BAD": "BAD",
                              "GOOD": "GOOD",
                              "MARGINAL": "MARGINAL",
                              "NONE": "NONE",
                            }
                          }
                          enumKeysToDisplayStrings={
                            Map {
                              "GOOD" => "Good",
                              "MARGINAL" => "Marginal",
                              "BAD" => "Bad",
                              "NONE" => "None",
                            }
                          }
                          enumToCheckedMap={
                            Map {
                              "GOOD" => true,
                              "BAD" => true,
                              "MARGINAL" => true,
                              "NONE" => true,
                            }
                          }
                          enumToColorMap={
                            Map {
                              "GOOD" => "#fff",
                              "MARGINAL" => "#fff",
                              "BAD" => "#fff",
                              "NONE" => "NULL_CHECKBOX_COLOR_SWATCH",
                            }
                          }
                          onChange={[Function]}
                        />
                      }
                      tooltip="Filter by Status"
                      widthPx={200}
                    >
                      <div
                        data-cy="filter-soh"
                      >
                        <Blueprint3.Button
                          active={false}
                          alignText="left"
                          className="toolbar-button"
                          onClick={[Function]}
                          style={
                            Object {
                              "width": "200px",
                            }
                          }
                          title="Filter by Status"
                        >
                          <button
                            className="bp3-button bp3-align-left toolbar-button"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            style={
                              Object {
                                "width": "200px",
                              }
                            }
                            title="Filter by Status"
                            type="button"
                          >
                            <Blueprint3.Icon
                              key="leftIcon"
                            />
                            <span
                              className="bp3-button-text"
                              key="text"
                            >
                              <span>
                                Filter by Status
                              </span>
                              <Blueprint3.Icon
                                icon="chevron-down"
                                title={false}
                              >
                                <span
                                  className="bp3-icon bp3-icon-chevron-down"
                                  icon="chevron-down"
                                >
                                  <svg
                                    data-icon="chevron-down"
                                    height={16}
                                    viewBox="0 0 16 16"
                                    width={16}
                                  >
                                    <path
                                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                      fillRule="evenodd"
                                      key="0"
                                    />
                                  </svg>
                                </span>
                              </Blueprint3.Icon>
                            </span>
                            <Blueprint3.Icon
                              key="rightIcon"
                            />
                          </button>
                        </Blueprint3.Button>
                      </div>
                    </PopoverButtonComponent>
                  </div>
                </Component>
              </div>
            </div>
            <div
              className="toolbar__center-group"
            >
              <div
                className="toolbar__whitespace"
                style={
                  Object {
                    "width": "94px",
                  }
                }
              />
            </div>
            <div
              className="toolbar__right-group"
            >
              <div
                className="toolbar-item
                    "
                key="1"
              >
                <Component
                  addToPopoverMap={[Function]}
                  hasIssue={false}
                  item={
                    Object {
                      "dropdownOptions": Object {
                        "CHANNEL_FIRST": "Channel: A - Z",
                        "CHANNEL_LAST": "Channel: Z - A",
                        "LAG_HIGHEST": "Lag: highest to lowest",
                        "LAG_LOWEST": "Lag: lowest to highest",
                      },
                      "label": "Sort Channels",
                      "onChange": [MockFunction],
                      "rank": 1,
                      "tooltip": "Sort",
                      "type": "Dropdown",
                      "value": "Lag: highest to lowest",
                      "widthPx": 220,
                    }
                  }
                  key="1"
                >
                  <div
                    style={Object {}}
                  >
                    <DropDown
                      dropDownItems={
                        Object {
                          "CHANNEL_FIRST": "Channel: A - Z",
                          "CHANNEL_LAST": "Channel: Z - A",
                          "LAG_HIGHEST": "Lag: highest to lowest",
                          "LAG_LOWEST": "Lag: lowest to highest",
                        }
                      }
                      key="1"
                      onMaybeValue={[Function]}
                      title="Sort"
                      value="Lag: highest to lowest"
                      widthPx={220}
                    >
                      <HTMLSelect
                        onChange={[Function]}
                        style={
                          Object {
                            "minWidth": "220px",
                            "width": "220px",
                          }
                        }
                        title="Sort"
                        value="Lag: highest to lowest"
                      >
                        <div
                          className="bp3-html-select"
                        >
                          <select
                            multiple={false}
                            onChange={[Function]}
                            style={
                              Object {
                                "minWidth": "220px",
                                "width": "220px",
                              }
                            }
                            title="Sort"
                            value="Lag: highest to lowest"
                          >
                            <option
                              key="LAG_HIGHEST"
                              value="Lag: highest to lowest"
                            >
                              Lag: highest to lowest
                            </option>
                            <option
                              key="LAG_LOWEST"
                              value="Lag: lowest to highest"
                            >
                              Lag: lowest to highest
                            </option>
                            <option
                              key="CHANNEL_FIRST"
                              value="Channel: A - Z"
                            >
                              Channel: A - Z
                            </option>
                            <option
                              key="CHANNEL_LAST"
                              value="Channel: Z - A"
                            >
                              Channel: Z - A
                            </option>
                          </select>
                          <Blueprint3.Icon
                            icon="double-caret-vertical"
                          >
                            <span
                              className="bp3-icon bp3-icon-double-caret-vertical"
                              icon="double-caret-vertical"
                            >
                              <svg
                                data-icon="double-caret-vertical"
                                height={16}
                                viewBox="0 0 16 16"
                                width={16}
                              >
                                <desc>
                                  double-caret-vertical
                                </desc>
                                <path
                                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                                  fillRule="evenodd"
                                  key="0"
                                />
                              </svg>
                            </span>
                          </Blueprint3.Icon>
                        </div>
                      </HTMLSelect>
                    </DropDown>
                  </div>
                </Component>
              </div>
              <div
                className="toolbar-item
                    "
                key="2"
              >
                <Component
                  addToPopoverMap={[Function]}
                  hasIssue={false}
                  item={
                    Object {
                      "hasIssue": false,
                      "label": "Last Updated",
                      "rank": 2,
                      "style": Object {
                        "marginLeft": "1em",
                      },
                      "tooltip": "Most recent SOH data received",
                      "tooltipForIssue": "SOH data has not updated in over 30 seconds",
                      "type": "LabelValue",
                      "value": "-",
                      "widthPx": 400,
                    }
                  }
                  key="2"
                >
                  <div
                    className="toolbar-label-value"
                    style={
                      Object {
                        "marginLeft": "1em",
                      }
                    }
                  >
                    <Component
                      label="Last Updated"
                      tooltip="Most recent SOH data received"
                      value="-"
                    >
                      <div
                        className="label-value-container"
                      >
                        <div
                          className="label-value__label"
                        >
                          Last Updated: 
                        </div>
                        <div
                          className="label-value__value"
                          style={
                            Object {
                              "color": "",
                            }
                          }
                          title="Most recent SOH data received"
                        >
                          -
                        </div>
                      </div>
                    </Component>
                  </div>
                </Component>
              </div>
              <div
                className="toolbar-item
                    "
                key="3"
              >
                <Component
                  addToPopoverMap={[Function]}
                  hasIssue={false}
                  item={
                    Object {
                      "label": "Update Interval",
                      "rank": 3,
                      "tooltip": "Interval at which SOH data is processed",
                      "type": "LabelValue",
                      "value": "20 seconds",
                      "widthPx": 400,
                    }
                  }
                  key="3"
                >
                  <div
                    className="toolbar-label-value"
                    style={Object {}}
                  >
                    <Component
                      label="Update Interval"
                      tooltip="Interval at which SOH data is processed"
                      value="20 seconds"
                    >
                      <div
                        className="label-value-container"
                      >
                        <div
                          className="label-value__label"
                        >
                          Update Interval: 
                        </div>
                        <div
                          className="label-value__value"
                          style={
                            Object {
                              "color": "",
                            }
                          }
                          title="Interval at which SOH data is processed"
                        >
                          20 seconds
                        </div>
                      </div>
                    </Component>
                  </div>
                </Component>
              </div>
            </div>
          </div>
        </ToolbarComponent>
      </Component>
    </Component>
    <div
      className="soh-drill-down-station-label display-title"
    >
      1
      <div
        className="display-title__subtitle"
      >
        Current lag per channel
      </div>
    </div>
  </div>
</Component>
`;
